### PR TITLE
feat(): #137 validate argument types of test()

### PR DESCRIPTION
**Pull request by adico1**

### CODE REVIEW

**Comment by adico1** [comment]
hi, the test fails because of this line:

```const aPendingTest = () => new Test('a work in progress', undefined, emptyTestCallbacks);```

in file: tests_factory.js, line 23.

The line creates a non valid test with body undefined.

What do you suggest?

**Comment by ngarbezza** [comment]
Oh, I just realized that a test with a name and no body, is considered pending. So we don't have a way to validate that :( I guess we can make sure the test body is either undefined or a valid function. Could you please change that? Sorry for the back and forth here.

File: lib/test.js [hunk 1]
```diff
@@ -103,6 +106,30 @@ class Test {
...
+  }
+  
+  _ensureNameIsValid(name) {
+    if (!isString(name)) {
+      throw 'Test does not have a valid name';
+    }
+  }
+  
+  _ensureBodyIsValid(body) {
+    if (!isFunction(body)) {
```
Comment (ngarbezza) [inline lib/test.js:129]: I guess the only thing that needs to change is to allow the `body` to be `undefined` as well, as it is a way to indicate that a test is pending, by no passing that argument:

`test('a WIP test')`

**Review by ngarbezza — COMMENTED** [review]
looks great @adico1 ! We just need a tiny change

**Review by ngarbezza — APPROVED** [review]
awesome! merging :rocket:

**Comment by ngarbezza** [comment]
@all-contributors let's thank @adico1 for his first code contribution and add them as a contributor! :tada:

**Comment by ngarbezza** [comment]
@all-contributors I mean, please add @adico1 for code :joy:

**Pull request merged**